src/main/java/org/embulk/input/mongodb/ValueCodec.java

```diff
@@ -53,8 +67,14 @@ public Value decode(final BsonReader reader, final DecoderContext decoderContext
             BsonType type = reader.getCurrentBsonType();
             fieldName = normalize(fieldName);
 
+            Value value;
             try {
-                kvs.put(newString(fieldName), readValue(reader, decoderContext));
+                value = readValue(reader, decoderContext);
+                kvs.put(newString(fieldName), value);
+                if (incrementalField.isPresent() && incrementalField.get().contains(fieldName)) {
```

Comment:
@sakama If we use both `id_field_name` and `incremental_field` and `incremental_filed` include `_id` field, this code is not working well. I think we should store `originalFIeldName` and use it here.